src/main.rs

```diff
@@ -98,9 +98,30 @@ async fn try_main() -> eyre::Result<bool> {
     // Set up the HTTP client
     let connect_timeout = Duration::from_secs(10);
     let timeout = Duration::from_secs(30);
-    let client = Client::builder()
+    let mut client_builder = Client::builder()
         .connect_timeout(connect_timeout)
-        .timeout(timeout)
+        .timeout(timeout);
+
+    // Add proxy if provided
+    let proxy_opt = config
+        .rsspls
+        .proxy
+        .or(env::var("http_proxy").ok())
+        .or(env::var("HTTPS_PROXY").ok());
```

Comment:
This works for now but I think the intention with these env vars is that they can both be set and point to different proxy servers.

Reply:
I don't understand how this works in conjunction, should it decide to use a certain one depending on the request itself being http/https?

Reply:
I think it's a remnant of the early internet where you could have different proxies for different protocols. Here's Netscape on Mac OS 8.1

<img width="445" alt="proxy-settings" src="https://github.com/wezm/rsspls/assets/21787/a1b27b3b-61ca-4e04-a9cc-502da6273e9c">

So yes, it would direct the request to depending on the protocol of the request. The example on https://docs.rs/reqwest/latest/reqwest/struct.Proxy.html demonstrates this: It uses `Proxy::http` with a `https` proxy url to "intercept all HTTP requests, and make use of the proxy at https://secure.example".

Reply:
Oh I see! Thanks for the interesting context, let me know if you think it's useful/needed and I could open another PR. I personally don't need it for my use.

Reply:
I think I’ll tweak it to match those semantics before publishing a release but no expectation for you to do it. I can do it.